src/components/pagination.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+function getPageRange(size) {
+  return [...Array(size).keys()].map((i) => i + 1);
+}
+
+class Pagination extends React.Component {
+  static propTypes = {
+    page: PropTypes.number,
+    pages: PropTypes.number,
+    onPageChange: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      page: props.page,
+      pages: props.pages,
+      onPageChange: props.onPageChange
+    };
+  }
+
+  componentDidUpdate(prevProps) {
```

Comment:
If these are not nested objects or need a deep-equal check. You can extend from `React.PureComponent` instead